src/main/java/org/vietabroader/view/MainView.java

```diff
@@ -47,6 +47,15 @@ private TitledPanel(String title) {
                     new EmptyBorder(8, 2, 8, 2))    // Inner padding of each panel
             );
         }
+
+        @Override
+        public void setEnabled(boolean b) {
+            this.setEnabled(b);
```

Comment:
use `super.setEnabled()`. Calling `this.setEnabled()` means calling this method, not the parent's one, making an indefinite recursive loop.